Source/Relativity.Test.Helpers/Relativity.Test.Helpers/UserHelpers/CreateUser.cs

```diff
@@ -14,234 +14,276 @@ public static class CreateUser
-
 			try
 			{
-				createResults = client.Repositories.User.Create(userDto);
+				const string errorContext = "An error occured when creating a new Relativity User.";
+				var userArtifactId = 0;
+				var firstName = "Temp User";
+				var randomGuid = Convert.ToString(Guid.NewGuid());
+				var lastName = randomGuid;
+				var emailAddress = "tempuser." + randomGuid + "@test.com";
+
+				//Code Types
+				const int defaultSelectedFileType = 1;
+				const int userType = 3;
+				const int documentSkip = 1000003;
+				const int skipDefaultPreference = 1000004;
+				const int password = 1000005;
+				const int sendNewPasswordTo = 1000006;
+
+				//Get the ArtifactIDs for the required Choice, Group, and Client objects.
+				var returnPasswordCodeId = FindChoiceArtifactId(client, sendNewPasswordTo, "Return");
+				var passwordCodeId = FindChoiceArtifactId(client, password, "Auto-generate password");
+				var documentSkipCodeId = FindChoiceArtifactId(client, documentSkip, "Enabled");
+				var documentSkipPreferenceCodeId = FindChoiceArtifactId(client, skipDefaultPreference, "Normal");
+				var defaultFileTypeCodeId = FindChoiceArtifactId(client, defaultSelectedFileType, "Native");
+				var userTypeCodeId = FindChoiceArtifactId(client, userType, "Internal");
+				var everyoneGroupArtifactId = FindGroupArtifactId(client, "Everyone");
+				var clientArtifactId = FindClientArtifactId(client, "Relativity Template");
+
+				var userDto = new User
+				{
+					AdvancedSearchPublicByDefault = true,
+					AuthenticationData = "",
+					BetaUser = false,
+					ChangePassword = true,
+					ChangePasswordNextLogin = false,
+					ChangeSettings = true,
+					Client = new Client(clientArtifactId),
+					DataFocus = 1,
+					DefaultSelectedFileType = new Choice(defaultFileTypeCodeId),
+					DocumentSkip = new Choice(documentSkipCodeId),
+					EmailAddress = emailAddress,
+					EnforceViewerCompatibility = true,
+					FirstName = firstName,
+					Groups = new List<kCura.Relativity.Client.DTOs.Group> { new kCura.Relativity.Client.DTOs.Group(everyoneGroupArtifactId) },
+					ItemListPageLength = 25,
+					KeyboardShortcuts = true,
+					LastName = lastName,
+					MaximumPasswordAge = 0,
+					NativeViewerCacheAhead = true,
+					PasswordAction = new Choice(passwordCodeId),
+					RelativityAccess = true,
+					SendPasswordTo = new Choice(returnPasswordCodeId),
+					SkipDefaultPreference = new Choice(documentSkipPreferenceCodeId),
+					TrustedIPs = "",
+					Type = new Choice(userTypeCodeId)
+				};
+
+				WriteResultSet<User> createResults;
+
+				try
+				{
+					createResults = client.Repositories.User.Create(userDto);
+				}
+				catch (Exception ex)
+				{
+					throw new TestHelpersException(errorContext, ex);
+				}
+
+				// Check for success.
+				if (createResults.Success)
+				{
+					userArtifactId = createResults.Results[0].Artifact.ArtifactID;
+				}
+				else
+				{
+					throw new TestHelpersException("Failed to Create User");

```

Comment:
i think createResults object has more error message which we can throw with the exception which would be helpful in troubleshooting.

Reply:
yes please throw the results of the createResults, not just the outer message, but each artifact's message if applicable.